OpenSim/Common/SmoothSegmentedFunction.cpp

```diff
@@ -39,6 +43,203 @@ static double UTOL = (double)SimTK::Eps*1e2;
 static double INTTOL = (double)SimTK::Eps*1e2;
 static int MAXITER = 20;
 static int NUM_SAMPLE_PTS = 100;
+
+//=============================================================================
+// PARAMETERS
+//=============================================================================
+
+/** Helper struct containing all parameters required to construct the
+ * SmoothSegmentedFunctionData.
+*/
+namespace OpenSim {
+struct SmoothSegmentedFunctionParameters {
+
+    SmoothSegmentedFunctionParameters(const SimTK::Matrix& mX,
+            const SimTK::Matrix& mY, double x0, double x1, double y0, double y1,
+            double dydx0, double dydx1, bool computeIntegral, bool intx0x1)
+            : _mX(mX), _mY(mY), _x0(x0), _x1(x1), _y0(y0), _y1(y1),
+              _dydx0(dydx0), _dydx1(dydx1), _computeIntegral(computeIntegral),
+              _intx0x1(intx0x1) {}
+
+    SimTK::Matrix _mX;
+    SimTK::Matrix _mY;
+
+    double _x0;
+    double _x1;
+    double _y0;
+    double _y1;
+    double _dydx0;
+    double _dydx1;
+    bool _computeIntegral;
+    bool _intx0x1;
+};
+
+bool operator==(
+    const SimTK::Matrix &lhs,
+    const SimTK::Matrix &rhs
+) {
+    if (lhs.nrow() != rhs.nrow() || lhs.ncol() != rhs.ncol()) {
+        return false;
+    }
+    for (int r = 0; r < lhs.nrow(); ++r)
+        for (int c = 0; c < lhs.ncol(); ++c)
+            if (lhs.row(r)[c] != rhs.row(r)[c]) return false;
+    return true;
+}
+
+bool operator==(
+    const SmoothSegmentedFunctionParameters &lhs,
+    const SmoothSegmentedFunctionParameters &rhs
+) {
+    return
+    lhs._x0 == rhs._x0 &&
```

Comment:
Performing the comparison out-of-order compared to the class definition is a premature optimization that makes it harder to match the list of elements here with the member data list (e.g. to see if something is missing from the equality calculation)